### PR TITLE
Ignore VMIs from backup when owner VMs are excluded

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**: 

This Pull Request aims to update the backup process for VMIs so that when the VMI is owned by a VM excluded from the backup, we also exclude the VMI without triggering an error.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://issues.redhat.com/browse/CNV-45094

**Special notes for your reviewer**:

**Release note**:

```release-note
Bugfix: Ignore VMIs from backup when owner VM is excluded
```

### CODE REVIEW

File: pkg/plugin/vmi_backup_item_action.go [hunk 1]
```diff
@@ -79,6 +79,15 @@ func (p *VMIBackupItemAction) Execute(item runtime.Unstructured, backup *v1.Back
 		return nil, nil, errors.WithStack(err)
 	}
 
+	// There's no point in backing up a VMI when it's owned by a VM excluded from the backup
```
[the remark this answers is not in the document]
Comment (alromeros) [reply pkg/plugin/vmi_backup_item_action.go:82]: I changed this to avoid triggering an error (for example, if launcher pod is excluded) if we eventually want to ignore the VMI. I like to have this as a self-contained check before doing anything else.

File: tests/resource_filtering_test.go [hunk 2]
```diff
@@ -2129,7 +2157,7 @@ var _ = Describe("Resource excludes", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 
-			It("[test_id:10227]Running VM excluded: backup should fail", func() {
+			It("[test_id:10227]Running VM excluded: empty backup without failure", func() {
```
[the remark this answers is not in the document]
Comment (alromeros) [reply tests/resource_filtering_test.go:2160]: I agree, I'd leave maybe two tests and delete the other two. I'm always hesitant to delete func tests though.

[the remark this answers is not in the document]
Comment (alromeros) [reply tests/resource_filtering_test.go:2160]: Also, are ok with deleting tests with test id?

[the remark this answers is not in the document]
Comment (ShellyKa13) [reply tests/resource_filtering_test.go:2160]: I think we have too many functional tests in this repo lol
And a lot of them are some edge cases which I dont think really interest anyone.. dont think it matters if it has test_id, just maybe the QE will need to take note that this tests are removed (same as to change the behavior of those you modified)

File: pkg/plugin/vmi_backup_item_action.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShellyKa13) [reply pkg/plugin/vmi_backup_item_action.go:82]: OK
